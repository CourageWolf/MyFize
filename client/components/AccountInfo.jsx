import React, { useState, useEffect } from 'react';

function AccountInfo (props) {

  console.log(props.accounts[0])




  return (
    <div className = "accInfo">
      <h3>Account Information</h3>
      <p><strong>Name: </strong>{props.accounts[0].account_name}</p>
      <p><strong>Type: </strong>{props.accounts[0].account_subtype}</p>
      <p><strong>Balance: </strong>${props.accounts[0].acount_balance}</p>
   </div>
  )
}


export default AccountInfo;