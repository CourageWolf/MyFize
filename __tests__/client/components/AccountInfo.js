import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

// import necessary components here
import AccountInfo from '../../../client/components/AccountInfo.jsx';
import { ExpansionPanelActions } from '@material-ui/core';

configure({ adapter: new Adapter() });

describe('AccountInfo React unit tests', () => {
  describe('AccountInfo', () => {
    let wrapper;
    const props = {
      accounts: [
        {
          account_id: 'yBDqZZbGZ5HX7GBrw654cpDGKWlP4ztyw43j8',
          account_subtype: 'checking',
          account_name: 'Plaid Gold Standard 0% Interest Checking',
          acount_balance: 100,
        },
        {
          account_id: '9B5rggwqgKHnekAZP8D4fPExkdlAryuRg4GAo',
          account_subtype: 'savings',
          account_name: 'Plaid Silver Standard 0.1% Interest Saving',
          acount_balance: 200,
        },
      ]
    };
    
    beforeAll(() => {
      wrapper = shallow(<AccountInfo {...props} />);
    });
    
    it('Should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    
    it('Should render a div with class of accInfo', () => {
      // expect(wrapper.find('div')).toBe(true);
      expect(wrapper.find('div').hasClass('accInfo')).toEqual(true);
    })
  });
});
  