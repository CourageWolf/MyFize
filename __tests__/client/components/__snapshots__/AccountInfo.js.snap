// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInfo React unit tests AccountInfo Should match snapshot 1`] = `
<div
  className="accInfo"
>
  <h3>
    Account Information
  </h3>
  <p>
    <strong>
      Name: 
    </strong>
    Plaid Gold Standard 0% Interest Checking
  </p>
  <p>
    <strong>
      Type: 
    </strong>
    checking
  </p>
  <p>
    <strong>
      Balance: 
    </strong>
    $
    100
  </p>
</div>
`;
